### DISCUSSION TITLE
автоматизация blockcheck.sh и install_easy.sh

### DISCUSSION THREAD

**Discussion by kotsmotritnastul** [bol-van/zapret#860] [Q&A]
стали очень часто меняться настройки на ТСПУ. может ли кто-нибудь сделать башскрипт который будет делать следующее:

1) останавливает запрет
2) запускает блокчек с сохранёнными настройками (адрес ггц, количество попыток, быстрый или стандартный тест и т.д.)
3) сохраняет предложенные варианты обхода блокировок в переменные или в текстовый файл
4) запускает ./install_easy.sh
5) подсовывает в него сохранённые параметры обхода блокировок
6) нажимает ввод во всех остальных случаях и запускает блокчек

такой скрипт можно было бы положить в кронтаб и запускать пока на работе работаешь, было бы очень удобно - сейчас после работы пара часов есть, и полчаса-час уходит на прогон блокчека потому что в рабочие часы они что-то опять поменяли. я пытался накостылить сам такое решение, но для этого надо менять имеющиеся баш-скрипты а они очень большие и я в них запутался.

**Comment by bol-van** [comment]
Здесь надо уходить от копипасты, записывать стратегии и находить общий знаменатель.
Может проблема банально в мешяющемся ttl

**Comment by kotsmotritnastul** [comment]
> Здесь надо уходить от копипасты, записывать стратегии и находить общий знаменатель. Может проблема банально в мешяющемся ttl

нет, меняются методы типа split/split2/fake и т.д. и кроме того на данный момент это всё работает, просто это надо делать руками, а это долго и муторно, возможно кто-нибудь возьмётся это автоматизировать.

**Comment by kotsmotritnastul** [comment]
> А можно вообще хотя бы узнать, что за параметры выдает блокчеккер и что с ними делать дальше? Потому что я не понимаю абсолютно НИЧЕГО, а в мануале про этот момент так же написано НИЧЕГО

я в своё время написал для себя и других памятку пошаговую https://gist.github.com/kotsmotritnastul/3cf8296a14d02bb95961c093082f41c0 люди вроде осиливают, может и тебе пригодится.

**Comment by kotsmotritnastul** [comment]
> Все равно ничего не понятно. Почему ты одни параметры записываешь в одну строку, а другие в другую? Я вот при подготовке блокчеккера везде энтер нажимал и мне выдало 4 строки в саммери. Как понять, какой параметр куда вписывать?

так в моей пошаговой инструкции же написано что куда пихать - параметр `curl_test_https_tls12` из блокчека относится к строке `NFQWS_OPT_DESYNC_HTTPS` в конфиге, а параметр `curl_test_http3` из блокчека относится к строке `NFQWS_OPT_DESYNC_QUIC` в конфиге. делай пошагово, по ней как минимум несколько десятков невдупляющих страдальцев делали и всё получилось.

**Reply by ait-ester** [reply]
Если честно, так и не понял, куда и в каком виде записывать параметры, которые мне порекомендовал zapret, даже более того - если запускать подробный анализ, выдача рекомендаций содержит порядка 12 листов (листов, не экранов!), соответственно - куда что распихивать в этом случае? И, кстати, строка `NFQWS_OPT_DESYNC_QUIC` в моём случае отсутствует, но есть просто
```
NFQWS_OPT="

--filter-tcp=80......
```
И ещё один момент: влияет ли на подбор параметров то, что я введу вместо `rr4---sn-5hne6nsd.googlevideo.com` просто `googlevideo.com`? Я правильно понимаю, что программа подбирает параметры под конкретный кеширующий сервер, и тогда проблемы могут возникнуть при смене отдающего сервера?

**Reply by numbereleven-a** [reply]
> Если честно, так и не понял, куда и в каком виде записывать параметры, которые мне порекомендовал zapret, даже более того - если запускать подробный анализ, выдача рекомендаций содержит порядка 12 листов (листов, не экранов!), соответственно - куда что распихивать в этом случае? И, кстати, строка `NFQWS_OPT_DESYNC_QUIC` в моём случае отсутствует, но есть просто
> 
> ```
> NFQWS_OPT="
> 
> --filter-tcp=80......
> ```
> 
> И ещё один момент: влияет ли на подбор параметров то, что я введу вместо `rr4---sn-5hne6nsd.googlevideo.com` просто `googlevideo.com`? Я правильно понимаю, что программа подбирает параметры под конкретный кеширующий сервер, и тогда проблемы могут возникнуть при смене отдающего сервера?

NFQWS_OPT_DESYNC - параметр из старой версии, сейчас --filter-tcp= актуально
если такое не понятно значит ридмишку не читал, ищи по там "Параметры nfqws"
подбор будет по *.googlevideo.com, но можно и под конкретный если указывать их в листе

**Comment by avatarDD** [comment]
У кого-то подписка на GPT-4o есть, надо его и попросить. Там дел минут на 10.

Вот, что gmini1.5 мне вывалил 🤣:

```bash
#!/bin/bash

# Проверка доступности YouTube через Selectel
check_youtube() {
    curl --connect-to ::speedtest.selectel.ru https://manifest.googlevideo.com/100MB -k -o /dev/null 2>&1
}

# Функция для обновления конфигурации zapret.conf
update_zapret_config() {
    local strategy="$1"
    local zapret_config="/etc/zapret/zapret.conf" # Предполагаемый путь к конфигу

    if [[ -z "$strategy" ]]; then
        echo "Ошибка: стратегия не указана"
        return 1
    fi

    if [[ ! -f "$zapret_config" ]]; then
        echo "Ошибка: файл конфигурации $zapret_config не найден"
        return 1
    fi

    # Обновляем конфигурацию (пример: добавляем стратегию в конец файла)
    echo "Обновление конфигурации zapret.conf с использованием стратегии: $strategy"
    sed -i "s/^STRATEGY=.*/STRATEGY=\"$strategy\"/" "$zapret_config"

    if [[ $? -eq 0 ]]; then
        echo "Конфигурация успешно обновлена"
    else
        echo "Ошибка обновления конфигурации"
        return 1
    fi
}

# Переменная для хранения результата
SUCCESS=0

# Основной процесс
if check_youtube; then
    echo "YouTube доступен"
else
    echo "YouTube недоступен. Начинаем подбор параметров..."

    # Остановка службы zapret
    echo "Останавливаем zapret..."
    systemctl stop zapret

    # Запуск сканирования blockcheck
    ./blockcheck.sh > blockcheck_output.txt

    # Извлечение всех стратегий из вывода blockcheck
    strategies=$(grep -E '^!!!!! curl_test_(http|https_tls12|https_tls13): working strategy found for ipv4 youtube.com : (.*) !!!!!$' blockcheck_output.txt | awk '{print $NF}')

    if [[ -z "$strategies" ]]; then
        echo "Не удалось найти подходящие стратегии в выводе blockcheck"
        exit 1
    fi

    echo "Найдены стратегии: $strategies"

    # Перебор стратегий
    for strategy in $strategies; do
        echo "Пробуем стратегию: $strategy"

        # Обновляем конфигурацию zapret
        if update_zapret_config "$strategy"; then
            # Перезапуск zapret
            echo "Запускаем zapret..."
            systemctl start zapret

            # Повторная проверка доступности YouTube
            if check_youtube; then
                echo "YouTube доступен после применения стратегии: $strategy"
                SUCCESS=1
                break
            else
                echo "Стратегия $strategy не помогла. Пробуем следующую..."
                systemctl stop zapret
            fi
        else
            echo "Ошибка обновления конфигурации zapret. Пропускаем стратегию $strategy."
        fi
    done

    if [[ $SUCCESS -eq 0 ]]; then
        echo "Не удалось восстановить доступ к YouTube. Проверьте настройки вручную."
    fi
fi
```
